fastai/learner.py

```diff
@@ -97,22 +97,22 @@ def before_epoch(self):
 class Learner(GetAttr):
     _default='model'
     def __init__(self,
-                 dls,  # `DataLoaders` containing data for each dataset needed for `model`
-                 model:callable, #  The model to train or use for inference
-                 loss_func:callable|None=None,  # Loss function for training
-                 opt_func=Adam,  # Optimisation function for training
-                 lr=defaults.lr,  # Learning rate
-                 splitter:callable=trainable_params,  # Used to split parameters into layer groups
-                 cbs=None,  # Callbacks
-                 metrics=None,  # Printed after each epoch
-                 path=None,  # Parent directory to save, load, and export models
-                 model_dir='models',  # Subdirectory to save and load models
-                 wd=None,  # Weight decay
-                 wd_bn_bias=False,  # Apply weight decay to batchnorm bias params?
-                 train_bn=True,  # Always train batchnorm layers?
-                 moms=(0.95,0.85,0.95),  # Momentum
-                 default_cbs:bool=True  # Include default callbacks?
-                ):
+        dls:DataLoaders, # `DataLoaders` containing fastai or PyTorch `DataLoader`s
+        model:callable|nn.Module, # PyTorch model for training or inference
+        loss_func:nn.Module|None=None, # Loss function. Defaults to `dls` loss
+        opt_func:Optimizer|OptimWrapper=Adam, # Optimization function for training
```

Comment:
I think (IIRC) any callable works, although not all features work with any callable. We should use the most general type which can work in some situation

Reply:
At minimum, `opt_func` needs to be a callable which accepts param groups from a splitter, and a learning rate per, `self.opt = self.opt_func(self.splitter(self.model), lr=self.lr)`, and then have a `step` implementation, etc. So I thought `Optimizer|OptimWrapper` was more useful than callable. Can change.

Reply:
Ah yes you're right - let's leave it as you have it then.

Reply:
Technically, it should be `Callable[...,Optimizer|OptimWrapper]|OptimWrapper`, but per fastai/nbdev#1179 this would probably cause `show_doc` to error out, hence my original suggestion.

Reply:
Right - we don't use generics.